src/async_impl/response.rs

```diff
@@ -69,6 +70,13 @@ impl Response {
         &mut self.headers
     }
 
+    /// Retrieve the cookies contained in the response.
+    pub fn cookies<'a>(&'a self) -> impl Iterator<
```

Comment:
I'm undecided, but what do you think of this mapping away the parse errors? So, if a response returns some cookies that are invalid, this iterator would just give the ones that are valid?

Reply:
I wondered about this too.

This api gives the most information but is the least convenient. 
An in-between solution would be `-> Result<impl Iterator<...>, Error> ` but I think nested generic `impl trait` does not work currently. 

Yeah you could always manually do `HeaderMap::get_all(SET_COOKIE)` and invalid cookie headers should be really rare.

Reply:
Ok I changed the result to be just `-> Iterator<Cookie..>`

Reply:
and added a comment to the methods about invalid headers being ignored